testing/sync-test/src/auth.rs

```diff
@@ -1,233 +1,46 @@
 /* Any copyright is dedicated to the Public Domain.
 http://creativecommons.org/publicdomain/zero/1.0/ */
 
-use crate::Opts;
 use anyhow::Result;
 use autofill::db::store::Store as AutofillStore;
-use fxa_client::internal::{auth, config::Config as FxaConfig, FirefoxAccount};
+use cli_support::fxa_creds::CliFxa;
+use fxa_client::internal::{config::Config as FxaConfig};
+use fxa_client::{Device, FirefoxAccount};
 use logins::LoginStore;
-use serde_json::json;
 use std::collections::HashMap;
 use std::sync::Arc;
-use sync15::DeviceType;
+use sync15::{client::{SetupStorageClient, Sync15StorageClient}, DeviceType};
 use sync_manager::{
-    manager::SyncManager, DeviceSettings, SyncAuthInfo, SyncEngineSelection, SyncParams, SyncReason,
+    manager::SyncManager, SyncEngineSelection, SyncParams, SyncReason, DeviceSettings
 };
 use tabs::TabsStore;
-use url::Url;
-use viaduct::Request;
 
-pub const CLIENT_ID: &str = "3c49430b43dfba77"; // Hrm...
-pub const SYNC_SCOPE: &str = "https://identity.mozilla.com/apps/oldsync";
+// Note that in revision ba9cc53710243c357e5bf125636bb94a7c6a4a48 or so, this had support for
+// username/password authentication and for "temporary" accounts using restmail etc.
+// It has since been refactored to use only oauth and to use the "cli-support" crate to help
+// manage the account, but you can use that revision if you ever want to revive username/password
+// support.
```

Comment:
Another thing we lose here (that we didn't use much in the first place) is the ability to fully automate the sync-tests where someone can run the sync-tests and it would go through the whole flow of creating the account, and syncing

Now the runner of the test would have to manually input the account information in a web browser window... given we no longer run the sync tests in CI I'm totally okay with it, it makes me a little sad that we couldn't get to a place where trust our integration tests enough to put them CI :(

Reply:
A bit sad, but I'm not sure that our repo is the place to test some of this stuff.  Account creation should probably be tested alongside the FxA server code.

If we want to prioritize it, we could re-add some sort of integration test as part of the FxA work.  I always wonder if it's worth it or not though.

Reply:
I think the point is more that now the tests can't run from a clean environment without user interaction - to complete an oauth flow. Before this change all you needed was an email and password and it was fully automated.